language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -730,32 +737,27 @@ instance ToJSON PackageJson where
 -- provided workspaces; write it back to disk.
 setupWorkspace :: FilePath -> FilePath -> [(T.Text, [Dependency])] -> IO ()
 setupWorkspace optInputPackageJson optOutputDir dependencies = do
-  -- If the file designated by 'optInputPackageJson' doesn't yet
-  -- exist, create it.
-  packageJsonExists <- doesFileExist optInputPackageJson
-  when (not packageJsonExists) $ do
-    BSL.writeFile optInputPackageJson $
-      encodePretty (object
-                    [ "private" .= True
-                    , "workspaces" .= ([] :: [T.Text])
-                    ])
   let (g, nodeFromVertex) = graphFromEdges'
         (map (\(a, ds) -> (a, a, map unDependency ds)) dependencies)
       ps = map (fst3 . nodeFromVertex) $ reverse (topSort g)
         -- Topologically order our packages.
       outBaseDir = T.pack $ takeFileName optOutputDir
         -- The leaf directory of the output directory (e.g. often 'daml2ts').
   let ourWorkspaces = map ((outBaseDir <> "/") <>) ps
-  bytes <- BSL.readFile optInputPackageJson
-  case decode bytes :: Maybe PackageJson of
-    Nothing -> fail $ "Error decoding JSON from '" <> optInputPackageJson <> "'"
-    Just oldPackageJson -> transformAndWrite ourWorkspaces outBaseDir oldPackageJson
+  mbBytes <- catchJust (guard . isDoesNotExistError)
+    (Just <$> BSL.readFile optInputPackageJson) (const $ pure Nothing)
+  packageJson <- case mbBytes of
+    Nothing -> pure mempty
```

Comment:
I'm not a big fan of using (highly) polymorphic constants in monomorphic code. What is `mempty` here? Is it `[]` or `Sum 0` or `Endo id`? I _guess_ it's an empty string of some sort. If so, I'd prefer to have the following for the sake of legibility:
```suggestion
    Nothing -> pure ""
```